core/src/main/java/com/oath/oak/OakAttachedWriteBuffer.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2018 Oath Inc.
+ * Licensed under the terms of the Apache 2.0 license.
+ * Please see LICENSE file in the project root for terms.
+ */
+
+package com.oath.oak;
+
+import java.nio.ByteBuffer;
+
+// An instance of OakWBufferScoped is only used when the write lock of the value referenced by it is already acquired.
+// This is the reason no lock is acquired in each access.
+class OakAttachedWriteBuffer extends OakAttachedReadBuffer implements OakWriteBuffer, OakUnsafeDirectBuffer {
+
+    OakAttachedWriteBuffer(Slice s, ValueUtils valueOperator) {
```

Comment:
This is exactly where we should have this "flag" or something to make it impossible to access out of the scope. Why isn't it here?

Reply:
The flag will be added only at the PR that will introduce the new API.